src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -12,6 +12,7 @@
 using System.Threading;
```

Comment:
@SteveL-MSFT, your last commit had 4 failures in `PowerShell-CI-linux`
SxS Module Path Basic Tests.ignore pshome module path derived from a different PowerShell instance
```powershell
Expected 3, but got 4.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Module/ModulePath.Tests.ps1: line 97
97:                 $paths.Count | Should -Be 3
```
TabCompletion.Should complete abbreviated function
```powershell
Expected exactly 'PSConsoleHostReadLine', but got @('PSConsoleHostReadLine', 'PSReadLine\PSConsoleHostReadLine').
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1: line 22
22:         $res | Should -BeExactly 'PSConsoleHostReadLine'
```
Import-Module.should be able to add a module with using ModuleInfo switch
```powershell
Expected exactly 'Microsoft.PowerShell.Security', but got @('Microsoft.PowerShell.Security', 'Microsoft.PowerShell.Security').
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 26
26:         (Get-Module -Name $moduleName).Name | Should -BeExactly $moduleName
```
Import-Module.should be able to load a module with a trailing directory separator: @('/home/vsts/work/1/a/bins/publish/Modules/Microsoft.PowerShell.Security', '/opt/microsoft/powershell/6/Modules/Microsoft.PowerShell.Security', /)
```powershell
Expected exactly 'Microsoft.PowerShell.Security', but got @('Microsoft.PowerShell.Security', 'Microsoft.PowerShell.Security').
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 26
26:         (Get-Module -Name $moduleName).Name | Should -BeExactly $moduleName
```